adagrams/game.py

```diff
@@ -1,11 +1,135 @@
+import random
+
 def draw_letters():
-    pass
+ 
+    NUM_TILES = 10
+    letters_grabbed = []
+    letters_remaining_pool = LIST_OF_LETTERS.copy()
+    
+    for i in range(NUM_TILES):
+        # update the last index of the remaining pool list that changes everytime
+        # random.randint stop is inclusive, so -1 which reflect the size of the pool
+        index = len(letters_remaining_pool)-1
+        # pop() allows to return the letter that was taken out and update the remaining pool list
+        letter_grabbed = letters_remaining_pool.pop(random.randint(0,index))
+        letters_grabbed.append(letter_grabbed)
+
+    return(letters_grabbed)
 
 def uses_available_letters(word, letter_bank):
-    pass
+
+    word = word.upper()
+    letter_bank_remaining = letter_bank.copy()
+
+    for letter in word:
+        # prevent the tiles that are used being used again ie "DOOR" cannot be made with D,O,R,X,X,X
+        if letter in letter_bank_remaining:
+            letter_bank_remaining.remove(letter)
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    
+    score = 0
+    word = word.upper()
+    for letter in word:
+        # score saved as values to the keys(letter)
+        score += VALUES_DICT[letter]
+    if len(word) >= 7:
+        score += 8
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    max_score = 0
+    max_word = ""
+
+    for word in word_list:
+        score = score_word(word)
+        # print(word, score, len(word))
+        
+        # if the score is greater than the current max, make it the new man
+        if score > max_score:
+            max_score = score
+            max_word = word
+        # if the scores are equal, have conditionals for tie breakers
+        elif score == max_score:
+            # if scores are the same but the existing max word lg is 10, or if theyre the same length
+            if len(max_word) == 10 or len(word) == len(max_word):
```

Comment:
We could use a constant variable to reference 10, maybe something like `TIE_BREAKER_LENGTH` or something else to describe why 10 is significant in this logic.